File: bank-account/src/main/java/org/suggs/sandbox/eventsourcing/bankaccount/accounts/bus/CommandBus.java
package org.suggs.sandbox.eventsourcing.bankaccount.accounts.bus;

import org.suggs.sandbox.eventsourcing.bankaccount.accounts.commandhandler.CommandHandler;
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.domain.commands.Command;

public interface CommandBus {

    void registerSubscriber(CommandHandler aCommand);

    void publish(Command aCommand);
}

File: bank-account/src/main/kotlin/org/suggs/sandbox/eventsourcing/bankaccount/accounts/bus/CommandBus.kt
package org.suggs.sandbox.eventsourcing.bankaccount.accounts.bus

import org.suggs.sandbox.eventsourcing.bankaccount.accounts.commandhandler.CommandHandler
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.domain.command.Command

interface CommandBus {

    fun registerSubscriber(aCommandHandler: CommandHandler<*>)

    fun publish(aCommand: Command)
}